bundles/org.eclipse.jface.text/src/org/eclipse/jface/internal/text/StickyHoverManager.java

```diff
@@ -142,6 +144,14 @@ public void stop() {
 			if (fInformationControlToClose != null)
 				fInformationControlToClose.removeFocusListener(this);
 
+			// Was added to fix the bug #2534
+			if (fInformationControlToClose instanceof AbstractInformationControl infoControl) {
+				Shell shell= infoControl.getShell();
+				if (!shell.isDisposed()) {
+					shell.setVisible(false);
```

Comment:
The fix works, but why don't we have that in `hideInformationControl()` ?
1) The "Closer" can be set so it is not guaranteed this code will work for all implementations.
2) The `Closer.stop()` javadoc says `Tells this closer to stop monitoring the subject and the information control` which doesn't fit here.

What about this patch:
```
diff --git a/bundles/org.eclipse.jface.text/src/org/eclipse/jface/text/AbstractInformationControlManager.java b/bundles/org.eclipse.jface.text/src/org/eclipse/jface/text/AbstractInformationControlManager.java
index d7be920..2146db2 100644
--- a/bundles/org.eclipse.jface.text/src/org/eclipse/jface/text/AbstractInformationControlManager.java
+++ b/bundles/org.eclipse.jface.text/src/org/eclipse/jface/text/AbstractInformationControlManager.java
@@ -26,2 +26,3 @@
 import org.eclipse.swt.widgets.Monitor;
+import org.eclipse.swt.widgets.Shell;
 
@@ -1212,4 +1213,11 @@
 			fInformationControl.setVisible(false);
-			if (fInformationControlCloser != null)
+			if (fInformationControlCloser != null) {
 				fInformationControlCloser.stop();
+			}
+			if (fInformationControl instanceof AbstractInformationControl infoControl) {
+				Shell shell= infoControl.getShell();
+				if (!shell.isDisposed()) {
+					shell.setVisible(false);
+				}
+			}
 		}
```

Reply:
Thanks.. Going to use this patch, test this and undo my changes.

Reply:
Please try. I have only quickly tested with Java editor, no idea if Xtext goes its own way, as I see many different implementations around.

Reply:
I've tested the solution in Java Editor and Xtext Editor.. Both are working.